core/amber/src/main/scala/edu/uci/ics/texera/web/resource/WorkflowWebsocketResource.scala

```diff
@@ -246,15 +278,29 @@ class WorkflowWebsocketResource {
 
   }
 
-  def killWorkflow(session: Session): Unit = {
-    WorkflowWebsocketResource.sessionJobs(session.getId)._2 ! PoisonPill
-    println("workflow killed")
+  /**
+    * clear the cache in the session when execute a new workflow
+    */
+  def clearCache(session: Session): Unit = {
+    sessionResults.remove(session.getId)
+    ResultDownloadResource.clearCache(session)
```

Comment:
I thought a workflow's download link response should be coupled within the `sessionResults`, or am I wrong?

Reply:
`sessionResults` is a hashmap in the service so I can not put more content into it. I have to create a new hashmap or save it into the session's user properties. I choose the later option. I think we should create a new service or data structure to handle all the data related to session instead of saving it as a new hashmap in the code.